lib/sla.js

```
@@ -34,62 +38,133 @@ function designateMixingBridge(client) {
-    defer.resolve(channel);
+  dialed.once('StasisStart', function(event, dialed) {
+    var answer = Q.denodeify(dialed.answer.bind(dialed));
+    answer();
+    defer.resolve(dialed);
+  });
+  dialed.once('ChannelDestroyed', function(event, dialed) {
+    defer.resolve(null);
+  });
+  inbound.once('ChannelHangupRequest', function(event, inbound) {
+    var hangup = Q.denodeify(dialed.hangup.bind(dialed));
+    hangup();
+    defer.resolve(null);
   });
   return defer.promise;
 }
 
+/** Simple utility function for checking if the bridgeName is numerical
+ * @param {String} bridgeName - the name of the bridge to be created/used
+ * @return {boolean} whether or not the bridgeName is numerical
+ */
+function checkIfNum (string) {
+  if(!isNaN(parseInt(string))) {
+    return true;
+  }
+  else {
+    return false;
+  }
+}
+
 /**
  * Receives initial input and begins the application.
  * @param {Object} client - Client received from app.js.
  * @return {Q} Q promise object.
  */
-module.exports = function(client) {
-  return designateMixingBridge(client)
-    .then(function (bridge) {
-      return createChannel(client, bridge)
-        .then(function (channel) {
-          return addChannelsToBridge(channel, bridge);
-        });
-    });
+module.exports = function(client, inbound, bridgeName) {
+  //This specifies what bridge number to use for SLA
+  if(checkIfNum(bridgeName)) {
+    var answer = Q.denodeify(inbound.answer.bind(inbound));
+    return answer()
+      .then(function() {
+        return designateMixingBridge(client, bridgeName);
+      }).then(function (bridge) {
+        return originateChannel(client, inbound, bridge)
+          .then(function (dialed) {
+            if (dialed) {
+              return addChannelsToBridge(inbound, dialed, bridge);
+            }
+            else {
+              var hangup = Q.denodeify(inbound.hangup.bind(inbound));
+              hangup();
+              return 'Channel rejected or caller hung up';
```

Returning a string for this is confusing. You should return a typed Error object. The caller can check the type to determine what to do with it. In other words, the program doesn't have to blow up, but returning a status as a string is error prone.